lib/screens/profile/profile.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:ecellapp/core/utils/logger.dart';
+import 'package:ecellapp/models/user.dart';
+import 'package:ecellapp/screens/profile/cubit/profile_cubit.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ProfileScreen extends StatefulWidget {
+  @override
+  _ProfileScreenState createState() => _ProfileScreenState();
+}
+
+class _ProfileScreenState extends State<ProfileScreen> {
+  User user;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BlocConsumer<ProfileCubit, ProfileState>(
+        listener: (context, state) {
+          if (state is ProfileError) {
+            Scaffold.of(context).showSnackBar(
+              SnackBar(content: Text(state.message)),
+            );
+          }
+        },
+        builder: (context, state) {
+          if (state is ProfileSuccess) {
+            user = state.user;
+            return _buildSuccess(context);
+          } else if (state is ProfileLoading) {
+            _profile();
+            return _buildLoading();
+          } else {
+            Log.e(tag: "ProfileState", message: "State now is ProfileError reached");
+            return Container(); // TODO link the homeScreen here
```

Comment:
show an error message on screen, like something went wrong. See, if an error occurred on this screen, there is nothing else to show, so the user should be shown the error on screen instead of a snackbar, and a retry button.

Reply:
Done!